### PR TITLE
fix: add retries to dataset pipeline for failure and success handlers.

### PR DESCRIPTION
**Pull request by Bento007**
- #3992 
- 
### Reviewers
**Functional:** @ebezzi 

**Readability:** @nayib-jose-gloria 

---

## Changes
- add retries to dataset pipeline for failure and success handlers.

### CODE REVIEW

File: .happy/terraform/modules/sfn/main.tf [hunk 1]
```diff
@@ -162,13 +162,26 @@ resource "aws_sfn_state_machine" "state_machine" {
...
         "Resource": "${var.lambda_success_handler}",
-        "End": true
+        "End": true,
+        "Retry": [ {
+            "ErrorEquals": ["Lambda.AWSLambdaException"],
+            "IntervalSeconds": 1,
+            "MaxAttempts": 3,
+            "BackoffRate": 2.0
+          }]
+        }
```
Comment (nayib-jose-gloria) [inline .happy/terraform/modules/sfn/main.tf:172]: I think this might be one curly brace too many, HandleErrors below should still be inside 'States'

**Review by nayib-jose-gloria — COMMENTED** [review]
one note to double check curly braces, but looks good otherwise

File: .happy/terraform/modules/sfn/main.tf (hunk 1 above)
Comment (ebezzi) [reply .happy/terraform/modules/sfn/main.tf:172]: If you update an rdev stack with this it should validate this code for you

**Comment by Bento007** [comment]
Removed a stray curly brace

File: .happy/terraform/modules/sfn/main.tf (hunk 1 above)
Comment (Bento007) [reply .happy/terraform/modules/sfn/main.tf]: @ebezzi it does work

**Review by nayib-jose-gloria — APPROVED** [review]

**Pull request merged**